pkg/ipamd/ipamd.go

```diff
@@ -652,7 +652,9 @@ func (c *IPAMContext) updateIPPoolIfRequired(ctx context.Context) {
 	log.Debugf("IP stats - total IPs: %d, assigned IPs: %d, cooldown IPs: %d", stats.TotalIPs, stats.AssignedIPs, stats.CooldownIPs)
 
 	if datastorePoolTooLow {
-		c.increaseDatastorePool(ctx)
+		for datastorePoolStillTooLow := datastorePoolTooLow; datastorePoolStillTooLow; datastorePoolStillTooLow, _ = c.isDatastorePoolTooLow() {
```

Comment:
nit: let's add a comment here explaining why we want to rapidly scale up, but we do not want to rapidly scale down